### PR TITLE
Publish pipeline with rsp bundling

### PR DESCRIPTION
**Pull request by jrichter1**
- bundle RSP from source
 - move staged builds to a new subfolder
 - add publish step

### CODE REVIEW

**Comment by jeffmaury** [comment]
I have an issue: rsp-server is a dependency like another for vscode-adapters so what's the reason for building it and not the anothers. Also at this pipeline is used for publishing to the marketplace, you will end building the current rsp-server and not a specific version.

**Comment by jrichter1** [comment]
I'm open to suggestions as to how to bundle the server. Right now it is just taken from the snapshots directory, which is the latest master build that runs once a day.

The only other dependency is the rsp-client, which is getting released through npm, so no need to build it manually. The server, however, I have no idea how the "stable" builds are being released yet. Once there is a consensus on that, I'll update this pipeline.

File: Jenkinsfile [hunk 1]
```diff
@@ -1,46 +1,83 @@
...
+	}
+
+	if(publishToMarketPlace.equals('true')) {
+		timeout(time:5, unit:'DAYS') {
+			input message:'Approve deployment?', submitter: 'jrichter'
+		}
+
+		stage("Publish to Marketplace") {
+			dir('adapters') {
+				withCredentials([[$class: 'StringBinding', credentialsId: 'vscode_java_marketplace', variable: 'TOKEN']]) {
```
Comment (mohitsuman) [inline Jenkinsfile]: @jrichter1 curious that `credentialsId: 'vscode_java_marketplace'`. Should we create one for adapters or use this.

Comment (jrichter1) [reply Jenkinsfile]: afaik these are the credentials for the Red Hat account in the marketplace, it's just that someone named them like this in jenkins

Comment (gorkem) [reply Jenkinsfile]: I would rather not rename, it is a variable name from back in the days where we were not planning to extend our presence. All plugins use the same and a rename will cause 4-5 jobs that needs to be fixed for no apparent gain.

**Pull request closed without merging**